.github/workflows/changelog.yml

```diff
@@ -0,0 +1,69 @@
+name: Generate changelog
+
+on:
+  workflow_dispatch:
+    inputs:
+
+      repository:
+        description: "Repository in which the changelog will be added."
+        required: false
+        default: 'aws'
+
+      ref:
+        description: |
+          Branch or ref to clone. This is where the changelog will be added and
+          ultimately committed.
+        required: true
+
+      type:
+        description: |
+          Type should be one of: feature, bugfix, enhancement, api-change
+          feature: A larger feature or change in behavior, usually resulting in a
+                   minor version bump.
+          bugfix: Fixing a bug in an existing code path.
+          enhancement: Small change to an underlying implementation detail.
+          api-change: Changes to a modeled API.
+        required: true
+
+      category:
+        description: |
+          Category is the high level feature area.
+          This can be a service identifier (e.g ``s3``),
+          or something like: Paginator.
+        required: true
+
+      description:
+        description: |
+          A brief description of the change.  You can
+          use github style references to issues such as
+          "fixes #489", "boto/boto3#100", etc.  These
+          will get automatically replaced with the correct
+          link.
+        required: true
+
+jobs:
+  add-changelog:
+    runs-on: Ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        repo: ${{ github.event.inputs.repo }}
```

Comment:
This key needs to be `repository` per the usage: https://github.com/actions/checkout#usage. I was seeing warnings as well about this in the sample workflow I ran.